### PR TITLE
Implement Borrowing List & Detail endpoint(test will be later because of relations with model Books)

### PR DESCRIPTION
**Pull request by oleksandra-shershen**
* Add borrowing model with constraints for borrow_date, expected_return_date, and actual_return_date.
* Implement a read serializer with detailed book info
* Implement list & detail endpoints

### CODE REVIEW

File: borrowing/models.py [hunk 1]
```diff
@@ -1,5 +1,42 @@
 from django.db import models
+from rest_framework.exceptions import ValidationError
+
+from library.models import Book
+from library_service import settings
```
Comment (Boyok1337) [inline borrowing/models.py:5]: `User = get_user_model()`

File: borrowing/models.py [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
+    borrow_date = models.DateField(auto_now_add=True)
+    expected_return_date = models.DateField()
+    actual_return_date = models.DateField(null=True, blank=True)
+    book = models.ForeignKey(
+        Book,
+        on_delete=models.CASCADE,
+        related_name="borrowings",
+    )
+    user = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
```
Comment (Boyok1337) [inline borrowing/models.py:17]: maybe better to use get_user_model()
initalize User = get_user_model()
and use User here

File: borrowing/serializers.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            "id",
+            "borrow_date",
+            "expected_return_date",
+            "actual_return_date",
+            "book",
+            "user"
+        )
+
+
+class BorrowingListSerializer(BorrowingSerializer):
```
Comment (Boyok1337) [inline borrowing/serializers.py]: u can use `serializers.ModelSerializer` if you define all variables, cause in future u can have conflict by inheritance, but if u use `serializers.ModelSerializer` u are independ on BorrowingSerializer

File: borrowing/serializers.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            "book",
+            "author",
+            "user",
+            "borrow_date",
+            "expected_return_date",
+            "actual_return_date",
+        )
+
+
+class BorrowingDetailSerializer(BorrowingListSerializer):
```
Comment (Boyok1337) [inline borrowing/serializers.py]: the same here

File: borrowing/views.py [hunk 5]
```diff
@@ -1,3 +1,24 @@
...
 
-# Create your views here.
+
+class BorrowViewSet(
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = Borrowing.objects.all()
```
Comment (Boyok1337) [inline borrowing/views.py]: remember to add optimization request to db

**Review by Boyok1337 — CHANGES_REQUESTED** [review]

File: borrowing/serializers.py (hunk 3 above)
Comment (oleksandra-shershen) [reply borrowing/serializers.py]: Resolved

File: borrowing/serializers.py (hunk 4 above)
Comment (oleksandra-shershen) [reply borrowing/serializers.py]: Resolved

File: borrowing/models.py (hunk 2 above)
Comment (oleksandra-shershen) [reply borrowing/models.py:17]: Using `settings.AUTH_USER_MODEL` in the ForeignKey field is generally preferred over using `get_user_model()`. I added this variable to `settings.py` with value `user.User`

File: borrowing/views.py (hunk 5 above)
Comment (oleksandra-shershen) [reply borrowing/views.py]: Forget this rule
Resolved

**Review by Gilgumish — APPROVED** [review]
GJ!

**Review by Boyok1337 — APPROVED** [review]
GJ!

**Pull request merged**